### PR TITLE
Allow setting both enable_autopilot and workload_identity_config in google_container_cluster resource

### PR DESCRIPTION
**Pull request by AshZhang**
This PR enables the creation of GKE clusters with both Autopilot and WorkloadALTS enabled (both `enable_autopilot` and `workload_alts_config` block set).

The [`workloadALTSConfig`](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.locations.clusters#workloadaltsconfig) JSON field in the GKE cluster REST resource requires the [`workloadIdentityConfig`](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.locations.clusters#workloadidentityconfig) field to be present as well, so the [`workload_alts_config`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster.html#nested_workload_alts_config) block requires the [`workload_identity_config`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster.html#nested_workload_identity_config) block to be present. This PR removes the `ConflictsWith` constraint between  [`enable_autopilot`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster.html#enable_autopilot)  and `workload_identity_config` so that a non-empty `workload_identity_config` block can be specified to satisfy the REST API to create the cluster.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: allowed both `enable_autopilot` and `workload_identity_config` to be set in `google_container_cluster` resource.
```

### CODE REVIEW

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb [hunk 1]
```diff
@@ -10032,6 +10105,30 @@ func testAccContainerCluster_withWorkloadALTSConfig(projectID, name, networkName
...
+		initial_node_count = 1
+		workload_alts_config {
+			enable_alts = %v
+		}
+		workload_identity_config {
+			workload_pool = "${data.google_project.project.project_id}.svc.id.goog"
+		}
+		enable_autopilot = true
+		deletion_protection = false
+	}
```
Comment (shuyama1) [inline mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb:10128]: only nit - we use spaces for indentation for Terraform configs

**Review by shuyama1 — COMMENTED** [review]
LGTM, only a small nitpick

File: mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb (hunk 1 above)
Comment (AshZhang) [reply mmv1/third_party/terraform/services/container/resource_container_cluster_test.go.erb]: Fixed.

**Review by shuyama1 — APPROVED** [review]
Thanks!

**Pull request merged**